infrastructure/src/main/java/org/corfudb/infrastructure/log/FileSystemAgent.java

```diff
@@ -71,7 +71,8 @@ private long getLogSizeLimit() {
 
         // Derived size in bytes that normal writes to the log unit are capped at.
         // This is derived as a percentage of the log's filesystem capacity.
-        return (long) (fileSystemCapacity * config.limitPercentage / 100);
+        long totalCorfuDataCapacity = fileSystemCapacity - (config.orthogonalData * 1024L);
+        return (long) (totalCorfuDataCapacity * config.limitPercentage / 100);
```

Comment:
You can get rid of the conversion if you change `orthogonalData` to a long and just accept the value in bytes.